client/chat/channel-status-card.tsx

```diff
@@ -0,0 +1,200 @@
+// shouldn't be used on its own.
+export function ChannelStatusCard({
+  channelName,
+  channelStatus,
+  isUserInChannel,
+  findChannelError,
+  joinChannelError,
+  onViewClick,
+  onJoinClick,
+}: ChannelStatusCardProps) {
+  const icon =
+    channelStatus && (!channelStatus.private || isUserInChannel) ? (
+      <StyledChannelIcon as={ChannelIcon} />
+    ) : (
+      <ErrorChannelIcon as={WarningIcon} />
+    )
+
+  const isChannelNotFound =
+    isFetchError(findChannelError) && findChannelError.code === ChatServiceErrorCode.ChannelNotFound
+  const isChannelClosed =
+    isFetchError(findChannelError) && findChannelError.code === ChatServiceErrorCode.ChannelClosed
+  const isUserBanned =
+    isFetchError(joinChannelError) && joinChannelError.code === ChatServiceErrorCode.UserBanned
+
+  let subtitle
+  if (isChannelNotFound) {
+    subtitle = <ErrorText>We couldn't find this channel, you may create it yourself</ErrorText>
+  } else if (isChannelClosed) {
+    subtitle = <ErrorText>We couldn't find this channel, it might have been closed</ErrorText>
+  } else if (channelStatus?.private && !isUserInChannel) {
+    subtitle = <ErrorText>This channel is private and requires an invite to join</ErrorText>
+  } else if (isUserBanned) {
+    subtitle = <ErrorText>You are banned from this channel</ErrorText>
+  } else if (channelStatus?.userCount) {
+    subtitle = (
+      <Body1>{`${channelStatus.userCount} member${channelStatus.userCount > 1 ? 's' : ''}`}</Body1>
+    )
+  }
+
+  let action
+  if (isUserInChannel) {
+    action = <RaisedButton label='View' onClick={onViewClick} />
+  } else if (isChannelNotFound) {
+    action = <RaisedButton label='Create' onClick={onJoinClick} />
+  } else if (channelStatus?.private || isUserBanned) {
+    action = <RaisedButton label='Join' disabled={true} />
+  } else if (!findChannelError) {
+    action = <RaisedButton label='Join' onClick={onJoinClick} />
+  }
+
+  return (
+    <Container>
+      {!channelStatus && !findChannelError ? (
+        <StyledLoadingDotsArea />
+      ) : (
+        <>
+          {icon}
+          <ChannelInfoContainer>
+            <Headline6>{channelName}</Headline6>
+            {subtitle}
+          </ChannelInfoContainer>
+          {action}
+        </>
+      )}
+    </Container>
+  )
+}
+
+export interface ConnectedChannelStatusCardProps {
+  channelId: SbChannelId
+  channelName: string
+}
+
+export function ConnectedChannelStatusCard({
+  channelId,
+  channelName,
+}: ConnectedChannelStatusCardProps) {
+  const dispatch = useAppDispatch()
+  const channel = useAppSelector(s => s.chat.byId.get(channelId))
+  const channelStatus = useAppSelector(s => s.channelStatus.byId.get(channelId))
+
+  const [findChannelError, setFindChannelError] = useState<Error>()
+  const [joinChannelError, setJoinChannelError] = useState<Error>()
+  const cancelJoinRef = useRef(new AbortController())
```

Comment:
imo this should probably be undefined unless you actually make a join request, but I'll admit I'm not really sure why you'd want to cancel an in-flight join request if the leave the page? If I click a join button and then navigate to a different channel I don't think I'm signifying that I don't want that to happen any more?